Nightly Dotty workflow of 2023-07-05 failed

See https://github.com/lampepfl/dotty/actions/runs/5460101159

```
 Caused by: java.util.NoSuchElementException: key not found: (com.vladsch.flexmark.html.HtmlRenderer$MainNodeRenderer@458e2fd5,Replace given matches by a library method)
Error:  	at scala.collection.MapOps.default(Map.scala:274)
Error:  	at scala.collection.MapOps.default$(Map.scala:273)
Error:  	at scala.collection.AbstractMap.default(Map.scala:405)
Error:  	at scala.collection.mutable.HashMap.apply(HashMap.scala:425)
Error:  	at dotty.tools.scaladoc.tasty.comments.markdown.SectionRenderingExtension$SectionHandler$.render(SectionRenderingExtension.scala:36)
Error:  	at dotty.tools.scaladoc.tasty.comments.markdown.SectionRenderingExtension$SectionHandler$.render(SectionRenderingExtension.scala:32)
Error:  	at com.vladsch.flexmark.html.renderer.NodeRenderingHandler.render(NodeRenderingHandler.java:16)
Error:  	at com.vladsch.flexmark.html.HtmlRenderer$MainNodeRenderer.renderNode(HtmlRenderer.java:779)
Error:  	at com.vladsch.flexmark.html.HtmlRenderer$MainNodeRenderer.renderChildrenNode(HtmlRenderer.java:798)
Error:  	at com.vladsch.flexmark.html.HtmlRenderer$MainNodeRenderer.renderChildren(HtmlRenderer.java:790)
Error:  	at com.vladsch.flexmark.html.renderer.CoreNodeRenderer.render(CoreNodeRenderer.java:113)
Error:  	at com.vladsch.flexmark.html.renderer.NodeRenderingHandler.render(NodeRenderingHandler.java:16)
Error:  	at com.vladsch.flexmark.html.HtmlRenderer$MainNodeRenderer.renderNode(HtmlRenderer.java:761)
Error:  	at com.vladsch.flexmark.html.HtmlRenderer$MainNodeRenderer.render(HtmlRenderer.java:683)
Error:  	at com.vladsch.flexmark.html.HtmlRenderer.render(HtmlRenderer.java:213)
Error:  	at com.vladsch.flexmark.html.HtmlRenderer.render(HtmlRenderer.java:197)
Error:  	at com.vladsch.flexmark.html.HtmlRenderer.render(HtmlRenderer.java:230)
Error:  	at dotty.tools.scaladoc.site.TemplateFile.resolveInner(templates.scala:128)
Error:  	at dotty.tools.scaladoc.site.LoadedTemplate.resolveToHtml(LoadedTemplate.scala:54)
Error:  	at dotty.tools.scaladoc.renderers.ResolvedTemplate.<init>(SiteRenderer.scala:18)
Error:  	at dotty.tools.scaladoc.renderers.ResolvedTemplate$.apply(SiteRenderer.scala:17)
Error:  	at dotty.tools.scaladoc.renderers.SiteRenderer.templateToPage(SiteRenderer.scala:25)
Error:  	at dotty.tools.scaladoc.renderers.SiteRenderer.templateToPage$(SiteRenderer.scala:21)
Error:  	at dotty.tools.scaladoc.renderers.Renderer.templateToPage(Renderer.scala:20)
Error:  	at dotty.tools.scaladoc.renderers.SiteRenderer.templateToPage$$anonfun$1(SiteRenderer.scala:26)
Error:  	at scala.collection.immutable.List.map(List.scala:250)
Error:  	at dotty.tools.scaladoc.renderers.SiteRenderer.templateToPage(SiteRenderer.scala:26)
Error:  	at dotty.tools.scaladoc.renderers.SiteRenderer.templateToPage$(SiteRenderer.scala:21)
Error:  	at dotty.tools.scaladoc.renderers.Renderer.templateToPage(Renderer.scala:20)
Error:  	at dotty.tools.scaladoc.renderers.SiteRenderer.templateToPage$$anonfun$1(SiteRenderer.scala:26)

```

--- a/scaladoc/src/dotty/tools/scaladoc/tasty/comments/markdown/SectionRenderingExtension.scala
+++ b/scaladoc/src/dotty/tools/scaladoc/tasty/comments/markdown/SectionRenderingExtension.scala
@@ -27,14 +27,15 @@ object SectionRenderingExtension extends HtmlRenderer.HtmlRendererExtension:
 
   case class AnchorLink(link: String) extends BlankLine(BasedSequence.EmptyBasedSequence())
   object SectionHandler extends CustomNodeRenderer[Section]:
-    val repeatedIds: mutable.Map[(NodeRendererContext, BasedSequence), Int] = mutable.Map()
+    val repeatedIds: mutable.Map[(NodeRendererContext, String), Int] = mutable.Map()
     val idGenerator = new HeaderIdGenerator.Factory().create()
     override def render(node: Section, c: NodeRendererContext, html: HtmlWriter): Unit =
       val Section(header, body) = node
-      val idSuffix = repeatedIds.getOrElseUpdate((c, header.getText), 0)
+      val headerText = header.getText.toString
+      val idSuffix = repeatedIds.getOrElseUpdate((c, headerText), 0)
       val ifSuffixStr = if(idSuffix == 0) then "" else idSuffix.toString
-      repeatedIds.update((c, header.getText), idSuffix + 1)
-      val id = idGenerator.getId(header.getText.append(ifSuffixStr))
+      repeatedIds.update((c, headerText), idSuffix + 1)
+      val id = idGenerator.getId(headerText + ifSuffixStr)
       val anchor = AnchorLink(s"#$id")
       val headerClass: String = header.getLevel match
         case 1 => "h500"

--- a/scaladoc/src/dotty/tools/scaladoc/tasty/comments/markdown/SectionRenderingExtension.scala
+++ b/scaladoc/src/dotty/tools/scaladoc/tasty/comments/markdown/SectionRenderingExtension.scala
@@ -33,7 +33,7 @@ object SectionRenderingExtension extends HtmlRenderer.HtmlRendererExtension:
       val Section(header, body) = node
       val idSuffix = repeatedIds.getOrElseUpdate((c, header.getText), 0)
       val ifSuffixStr = if(idSuffix == 0) then "" else idSuffix.toString
-      repeatedIds.update((c, header.getText), repeatedIds((c, header.getText)) + 1)
+      repeatedIds.update((c, header.getText), idSuffix + 1)
       val id = idGenerator.getId(header.getText.append(ifSuffixStr))
       val anchor = AnchorLink(s"#$id")
       val headerClass: String = header.getLevel match

